src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -528,54 +531,54 @@ void WellGeneratorBase::debugWellGeometry() const
+  TableLayout tableWellLayout = TableLayout( {
+      TableLayout::ColumnParam{"Element no.", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"CoordX", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordY", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordZ", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"Prev\nElement", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"Next\nElement", TableLayout::Alignment::right},
+    } );
+
+  tableWellLayout.setTitle( "InternalWellGenerator " + getName());
+
+  TableData tableWellData;
 
+  //2. collecting data
   for( globalIndex iwelem = 0; iwelem < m_numElems; ++iwelem )
   {
-    std::cout << "Well element #" << iwelem << std::endl;
-    std::cout << "Coordinates of the element center: " << m_elemCenterCoords[iwelem] << std::endl;
-    if( m_nextElemId[iwelem] < 0 )
-    {
-      std::cout << "No next well element" << std::endl;
-    }
-    else
-    {
-      std::cout << "Next well element # = " << m_nextElemId[iwelem] << std::endl;
-    }
-    if( m_prevElemId[iwelem][0] < 0 )
-    {
-      std::cout << "No previous well element" << std::endl;
-    }
-    else
+    std::optional< globalIndex > nextElement;
+    std::optional< globalIndex > prevElement;
+
+    if( m_nextElemId[iwelem] >= 0 )
     {
-      std::cout << "Previous well element #" << m_prevElemId[iwelem][0] << std::endl;
+      nextElement =  m_nextElemId[iwelem];
     }
-    for( globalIndex inode = 0; inode < m_numNodesPerElem; ++inode )
+
+    if( m_prevElemId[iwelem][0] >= 0 )
     {
-      if( inode == 0 )
-      {
-        std::cout << "First well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
-      else
-      {
-        std::cout << "Second well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
+      prevElement =  m_prevElemId[iwelem][0];
     }
+
+    tableWellData.addRow( iwelem, m_elemCenterCoords[iwelem][0], m_elemCenterCoords[iwelem][1], m_elemCenterCoords[iwelem][2], prevElement, nextElement );
   }
 
-  std::cout << std::endl << "Number of perforations = " << m_numPerforations << std::endl;
+  //3. dumping
+  TableTextFormatter tableFormatter( tableWellLayout );
+  GEOS_LOG_RANK_0( tableFormatter.ToString( tableWellData ));
 
+  TableLayout tableLayoutPerfo = TableLayout( {"Perforation no.", "Coordinates", "connected to" } );
+  tableLayoutPerfo.setTitle( "Peforation table" );
+
+  TableData tablePerfoData;
   for( globalIndex iperf = 0; iperf < m_numPerforations; ++iperf )
   {
-    std::cout << "Perforation #" << iperf << std::endl;
-    std::cout << "Coordinates of the perforation: " << m_perfCoords[iperf] << std::endl;
-    std::cout << "Is connected to well element #" << m_perfElemId[iperf] << std::endl;
+    tablePerfoData.addRow( iperf, m_perfCoords[iperf], m_perfElemId[iperf] );
   }
-  std::cout << std::endl;
+
+  TableTextFormatter tablePerfoLog( tableLayoutPerfo );
```

Comment:
I would have used scope brackets here as we clearly have two distincts portions of code (for well and perfo).
```c++
  { // well data output
    ...
  }

  { // perforations data output
    ...
  }
```